pallets/eth-bridge/src/lib.rs

```diff
@@ -651,6 +670,81 @@ pub mod pallet {
 
         Ok(())
     }
+    fn has_author_casted_event_vote<T: Config>(author: &T::AccountId) -> bool {
+        for (_partition, votes) in EthereumEvents::<T>::iter() {
```

Comment:
Since each collator can vote once, we iterate through a small number of elements.
If we use the partition data, that would allow a collator to cast as many votes as he wants, bloating the storage.